server/controllers/user.controller.js

```diff
@@ -78,7 +78,7 @@ function update(req, res, next) {
       user.avatarUrl = `https://${S3_BUCKET}.s3.amazonaws.com/${user._id}`
     }
     return user.save().then((newUser) => {
-      const userMinusPassword = Object.assign(newUser, {password: null});
+      const userMinusPassword = Object.assign({}, newUser, {password: null});
```

Comment:
Was this the bug that resulted in null passwords?

Reply:
This had been the area yeah but it looked like this:

user.save() // This is async so next line will execute before this completes, causing the bug
user.password = null
res.json(user);

Reply:
Wow, async bug, that is nasty.

Reply:
Yea, all mongo operations are promises except queries need to use exec.